Add Smart Display sample (#2649)
microsoft/BotFramework-WebChat: samples/01.d.getting-started-es5-direct-line-speech/README.md

> # Sample - Getting Started with Web Chat ES5 bundle with Direct Line Speech

Reviewer: ```suggestion
# Sample - Getting Started with Web Chat ES5 bundle and Direct Line Speech
```

## Description

> A simple web page with a maximized Web Chat that connect to Direct Line Speech channel and works under Internet Explorer 11.

Reviewer: ```suggestion
A simple web page with a maximized Web Chat that connects to the Direct Line Speech channel and works in Internet Explorer 11.
```

> Although the web page works under Internet Explorer 11, the speech recognition and synthesis will be disabled when running under Internet Explorer 11. When running the web page under a browser with required media capabilities, speech recognition and synthesis will be enabled/

Reviewer: ```suggestion
Although the web page works in Internet Explorer 11, the speech recognition and synthesis will be disabled when running in IE11. When running the web page under a browser with required media capabilities, speech recognition and synthesis will be enabled
```

# Test out the hosted sample

-  [Try out MockBot](https://microsoft.github.io/BotFramework-WebChat/01.d.getting-started-es5-direct-line-speech)

# How to run

-  Fork this repository
-  Navigate to `/Your-Local-WebChat/samples/01.d.getting-started-es5-direct-line-speech` in command line
-  Run `npx serve`
-  Browse to [http://localhost:5000/](http://localhost:5000/)

# Things to try out

> -  Type `hello`: you should see the initials displayed next to user and bot speech bubbles in the transcript.

Reviewer: ??

# Code

> Jump to [completed code](#completed-code) to see the end-result `index.html`.

## Getting started

### Goals of this bot

> This sample will show you how to implement your own initials displayed next to .

Reviewer: ??

This sample starts with the [ES5 bundle CDN sample](./../01.a.getting-started-es5-bundle/README.md) as the base template.

> First, we will remove the existing fetch token call with a `fetchCredentials` function, which asynchronously return a JavaScript object of `{ authorizationToken: string, region: string }` in a Promise fashion.

Reviewer: ```suggestion
First, we will remove the existing fetch token call with a `fetchCredentials` function, which asynchronously returns a JavaScript object of `{ authorizationToken: string, region: string }` in a Promise fashion.
```

```js
const fetchCredentials = function() {
  return window
    .fetch('https://webchat-mockbot-streaming.azurewebsites.net/speechservices/token', {
      method: 'POST'
    })
    .then(function(res) {
      return res.json();
    })
    .then(function(json) {
      return {
        authorizationToken: json.token,
        region: json.region
      };
    });
};
```

> Then, we will create a Direct Line Speech adapter set with the `fetchCredentials` function asynchronously. And use the adapter set to initialize Web Chat.

Reviewer: ```suggestion
Next we will create a Direct Line Speech adapter set with the `fetchCredentials` function asynchronously and use the adapter set to initialize Web Chat.
```

```diff
+ window.WebChat.createDirectLineSpeechAdapters({ fetchCredentials: fetchCredentials }).then(function(adapters) {
    window.WebChat.renderWebChat(
-     {
-       directLine: window.WebChat.createDirectLine({
-         token: token
-       })
-     },
+     Object.assign({}, adapters),
      document.getElementById('webchat')
    );

    document.querySelector('#webchat > *').focus();
  });
```

> If you need to pass options to Web Chat, replace the `{}` with your options.

## Completed code

Here is the finished `index.html`:

```html
<!DOCTYPE html>
<html lang="en-US">
  <head>
    <title>Web Chat: Avatar with images and initials</title>

>     <script src="https://cdn.botframework.com/botframework-webchat/latest/webchat.js"></script>

Reviewer: is this the right bundle?
